src/main.rs

```diff
@@ -230,49 +208,37 @@ fn main() {
         })
         .collect();
     let duration = start.elapsed();
-    println!(
-        "Found exhaustive neighbors for sample to calculate recall in {:?}",
-        duration
-    );
-    // Main parameters
-    let input_words = ["river", "war", "love", "education"];
-    let words_to_visualize: Vec<String> =
-        input_words.into_iter().map(|x| x.to_owned()).collect();
-    let index_results = build_benchmark_and_visualize_index::<DIM>(
-        &my_input_data,
-        &word_to_idx_mapping,
-        &idx_to_word_mapping,
-        3,
-        15,
-        TOP_K,
-        &sample_idx,
-        &words_to_visualize,
-    );
-    calculate_metrics_on_index_result::<DIM>(
-        &my_input_data,
-        &exhaustive_results,
-        &index_results,
+    println!("Found exhaustive neighbors for sample in {:?}", duration);
+    // Build, benchmark and visualize our index with default parameters
+    let words_to_visualize: Vec<String> = ["river", "war", "love", "education"].into_iter().map(|x| x.to_owned()).collect();
+    let index_results = build_and_benchmark_index::<DIM>(
+        &my_input_data, 3, 15, TOP_K,
     );
-    // Try some other parameters. New values for max_node_size, num_trees at dim=300. See how we can make it
-    // better in its accuracy/ quality.
-    let no_words: Vec<String> = Vec::new();
-    for num_trees in [3, 9, 15] {
-        for max_node_size in [5, 15, 30] {
-            let index_results = build_benchmark_and_visualize_index::<DIM>(
-                &my_input_data,
-                &word_to_idx_mapping,
-                &idx_to_word_mapping,
-                num_trees,
-                max_node_size,
-                TOP_K,
-                &sample_idx,
-                &no_words,
-            );
-            calculate_metrics_on_index_result::<DIM>(
-                &my_input_data,
-                &exhaustive_results,
-                &index_results,
-            );
-        }
-    }
+    // calculate_metrics_on_index_result::<DIM>(
+    //     &my_input_data,
+    //     &exhaustive_results,
+    //     &index_results,
+    // );
+    // // Try some other parameters. New values for max_node_size, num_trees at dim=300. See how we can make it
+    // // better in its accuracy/ quality.
+    // let no_words: Vec<String> = Vec::new();
+    // for num_trees in [3, 9, 15] {
+    //     for max_node_size in [5, 15, 30] {
+    //         let index_results = build_benchmark_and_visualize_index::<DIM>(
+    //             &my_input_data,
+    //             &word_to_idx_mapping,
+    //             &idx_to_word_mapping,
+    //             num_trees,
+    //             max_node_size,
+    //             TOP_K,
+    //             &sample_idx,
+    //             &no_words,
+    //         );
+    //         calculate_metrics_on_index_result::<DIM>(
+    //             &my_input_data,
+    //             &exhaustive_results,
+    //             &index_results,
+    //         );
+    //     }
```

Comment:
Do you plan to bring all this back? If not, just delete the code maybe?

Reply:
Brought it all back and made the rust file runnable end-to-end from the benchmarking script. All experimental code can now run including average euclidean and recall calculations